user/test/com/google/gwt/emultest/java11/lang/StringTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.emultest.java11.lang;
+
+import com.google.gwt.emultest.java.util.EmulTestBase;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * Tests for java.lang.String Java 11 API emulation.
+ */
+public class StringTest extends EmulTestBase {
+    public void testIsBlank() {
+        assertTrue("".isBlank());
+        assertTrue("  ".isBlank());
+        assertFalse("x ".isBlank());
+        assertTrue("\u001c".isBlank());
+        assertFalse("\u00a0".isBlank());
+    }
+
+    public void testStrip() {
+        assertEquals("", "".strip());
+        assertEquals("", "  ".strip());
+        assertEquals("x", " x ".strip());
+        assertEquals("x", "\u001cx\u001c".strip());
+        assertEquals("\u00a0x\u00a0", "\u00a0x\u00a0 ".strip());
+    }
+
+    public void testStripLeading() {
+        assertEquals("", "".stripLeading());
+        assertEquals("", "  ".stripLeading());
+        assertEquals("x ", " x ".stripLeading());
+        assertEquals("x\u001c", "\u001cx\u001c".stripLeading());
+        assertEquals("\u00a0x\u00a0", "\u00a0x\u00a0".stripLeading());
+    }
+
+    public void testStripTrailing() {
+        assertEquals("", "".stripTrailing());
+        assertEquals("", "  ".stripTrailing());
+        assertEquals(" x", " x ".stripTrailing());
+        assertEquals("\u001cx", "\u001cx\u001c".stripTrailing());
+        assertEquals("\u00a0x\u00a0", "\u00a0x\u00a0 ".stripTrailing());
+    }
+
+    public void testRepeat() {
+        assertEquals("", "foo".repeat(0));
+        assertEquals("foo", "foo".repeat(1));
+        assertEquals("foofoofoo", "foo".repeat(3));
+        try {
+            String noFoo = "foo".repeat(-1);
+            throw new Error("Should fail with negative arg");
+        } catch (IllegalArgumentException ex) {
+            assertEquals("count is negative: -1", ex.getMessage());
+        }
+    }
+
+    public void testLines() {
+        assertEquals(Arrays.asList("a", "b", "c", "d"),
+                "a\rb\nc\r\nd".lines().collect(Collectors.toList()));
```

Comment:
Worth testing a \n\r case (that is, the opposite of \r\n)? I think you've got this right, but can't be sure just from looking at it. \r\n should be one line separator, but \n\r should be two. Expanding on this \r\n\r and \n\r\n should be two, but \r\n\r\n should be two while \n\r\n\r should be three (\n, \r\n, \r)